src/python/essentia/pytools/batch_extractors.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2006-2020  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+from argparse import ArgumentParser
+from multiprocessing import Pool
+from multiprocessing import cpu_count
+from subprocess import Popen, PIPE
+from essentia import EssentiaError
+from functools import partial
+import os
+import sys
+
+
+def __subprocess(cmd, verbose=True):
+    """General purpose subprocess.
+    """
+
+    cmd_str = ' '.join([str(x) for x in cmd])
+
+    child = Popen(cmd, stdout=None, stderr=PIPE)
+    _, stderr = child.communicate()
+    rc = child.returncode
+
+    if verbose:
+        if rc == 0:
+            print('"{}"... ok!'.format(cmd_str))
+        else:
+            print('"{}"... failed!'.format(cmd_str))
+            print(stderr, '\n')
+
+
+    return rc, cmd_str, stderr.decode("utf-8")
+
+
+def __batch_extractor(audio_dir, output_dir, extractor_cmd, output_extension,
+                      generate_log=True, audio_types=None, skip_analyzed=False,
+                      jobs=0, verbose=True):
+    if not audio_types:
+        audio_types = ('.wav', '.aiff', '.flac', '.mp3', '.ogg')
+        print("Audio files extensions considered by default: " +
+              ' '.join(audio_types))
+    else:
+        if type(audio_types) == str:
+            audio_types = [audio_types]
+
+        audio_types = tuple(audio_types)
+        audio_types = tuple([i.lower() for i in audio_types])
+        print("Searching for audio files extensions: " + ' '.join(audio_types))
+    print("")
+
+    output_dir = os.path.abspath(output_dir)
+    audio_dir = os.path.abspath(audio_dir)
+
+    if jobs == 0:
+        try:
+            jobs = cpu_count()
+        except NotImplementedError:
+            print('Failed to automatically detect the cpu count, '
+                  'the analysis will try to continue with 4 jobs. '
+                  'For a different behavior change the `job` parameter.')
+            jobs = 4
+
+    skipped_count = 0
+    skipped_files = []
+    cmd_lines = []
+    for root, _, filenames in os.walk(audio_dir):
+        for filename in filenames:
+            if filename.lower().endswith(audio_types):
```

Comment:
wow, I didn't know this feature! interesting!
I've also done it by doing `if os.path.splitext(filename.lower())[1] in audio_types`